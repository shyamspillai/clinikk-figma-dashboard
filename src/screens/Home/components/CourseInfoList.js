import { useState, useEffect } from 'react';
import { Space } from 'antd'

import CourseInfo from './CourseInfo';
import { courseCategories } from '../../../data/categoryData';
import { generateFakeCourses } from '../../../data/courseData';

import styled from 'styled-components'

const CourseInfoList = () => {

    const [currentCategory, setCategory] = useState(0);
    const [courseCatalog, setCourseCatalog] = useState(generateFakeCourses());

    useEffect(() => {
        setCourseCatalog(generateFakeCourses())
    }, [currentCategory])

    const showCourseCategories = () => {
        return courseCategories.map((courseCategory, index) => <TabLink
            isActive={index === currentCategory}
            onClick={() => setCategory(index)}
            key={index}
            data-test={`catalog-${index}`}
            className="baloo weight-600 size-16">
            {courseCategory.title}
        </TabLink>
        );
    }

    const showCourseCatalog = () => {
        return courseCatalog.map(({ title, author, time, rating, icon }, index) => {
            return <CourseInfo
                data-test={`course-${index}`}
                icon={icon} title={title}
                author={author}
                time={time}
                rating={rating}
                key={index}
            />;
        })
    }

    return (
        <>
            <div className="flex row" >
                <Space size="large">
                    {showCourseCategories()}
                </Space>
            </div>

            <div className="flex column">
                {showCourseCatalog()}
            </div>
        </>
    )
}

export const TabLink = styled.span`
    color: ${props => props.isActive ? '#000000' : '#BFBFBF'};
    cursor: ${props => props.isActive ? 'default' : 'pointer'}
`

export default CourseInfoList
