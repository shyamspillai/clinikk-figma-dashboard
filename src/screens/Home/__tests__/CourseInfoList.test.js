import React from 'react';
import CourseInfoList from '../components/CourseInfoList';

import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe('Course List component', () => {

  it('should match snapshot', () => {
    const CourseInfoListComponent = shallow(<CourseInfoList />);
    expect(toJson(CourseInfoListComponent)).toMatchSnapshot();
  })

  it('should render the course info list correctly', () => {
    const wrapper = shallow(
      <CourseInfoList />
    );

    expect(wrapper.find("[data-test='catalog-0']").text()).toBe("All Courses");
    expect(wrapper.find("[data-test='catalog-1']").text()).toBe("The Newest");
    expect(wrapper.find("[data-test='catalog-2']").text()).toBe("Top Rated");
    expect(wrapper.find("[data-test='catalog-3']").text()).toBe("Most Popular");

    expect(wrapper.find("[data-test='course-0']").exists()).toBe(true)
    expect(wrapper.find("[data-test='course-1']").exists()).toBe(true)
    expect(wrapper.find("[data-test='course-2']").exists()).toBe(true)
    expect(wrapper.find("[data-test='course-3']").exists()).toBe(true)
    expect(wrapper.find("[data-test='course-4']").exists()).toBe(true)

  });

});