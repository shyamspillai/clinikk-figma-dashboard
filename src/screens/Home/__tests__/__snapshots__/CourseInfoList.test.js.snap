// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course List component should match snapshot 1`] = `
<Fragment>
  <div
    className="flex row"
  >
    <Space
      size="large"
    >
      <styled.span
        className="baloo weight-600 size-16"
        data-test="catalog-0"
        isActive={true}
        key="0"
        onClick={[Function]}
      >
        All Courses
      </styled.span>
      <styled.span
        className="baloo weight-600 size-16"
        data-test="catalog-1"
        isActive={false}
        key="1"
        onClick={[Function]}
      >
        The Newest
      </styled.span>
      <styled.span
        className="baloo weight-600 size-16"
        data-test="catalog-2"
        isActive={false}
        key="2"
        onClick={[Function]}
      >
        Top Rated
      </styled.span>
      <styled.span
        className="baloo weight-600 size-16"
        data-test="catalog-3"
        isActive={false}
        key="3"
        onClick={[Function]}
      >
        Most Popular
      </styled.span>
    </Space>
  </div>
  <div
    className="flex column"
  >
    <CourseInfo
      author="Zora White"
      data-test="course-0"
      icon={<ForwardRef(RedditSquareFilled) />}
      key="0"
      rating="9.04"
      time="3h 8min"
      title="Learn Javascript"
    />
    <CourseInfo
      author="Samir Rodriguez"
      data-test="course-1"
      icon={<ForwardRef(FacebookOutlined) />}
      key="1"
      rating="0.41"
      time="2h 7min"
      title="Learn Figma"
    />
    <CourseInfo
      author="Ova Crona"
      data-test="course-2"
      icon={<ForwardRef(ChromeOutlined) />}
      key="2"
      rating="7.45"
      time="1h 8min"
      title="Learn Travel Writing"
    />
    <CourseInfo
      author="Ivy Watsica"
      data-test="course-3"
      icon={<ForwardRef(TwitterOutlined) />}
      key="3"
      rating="7.45"
      time="6h 7min"
      title="Learn Skating"
    />
    <CourseInfo
      author="Nya Kunde"
      data-test="course-4"
      icon={<ForwardRef(IeOutlined) />}
      key="4"
      rating="7.31"
      time="1h 2min"
      title="Learn Artifial Intelligence"
    />
  </div>
</Fragment>
`;
